integration-tests/scripts/baseline/main.py

```diff
@@ -209,19 +250,29 @@ def process(content):
         {
             "kernel": record.get("VmConfig").replace('_', '.'),
             "collection_method": normalize_collection_method(record.get("CollectionMethod")),
-            "timestamp": record.get("Timestamp"),
-            record["TestName"]: record.get("Metrics").get("hackbench_avg_time")
+            "timestamp": stats.get("Timestamp"),
+            "test": record.get("TestName"),
+            "cpu": stats.get("Cpu"),
+            "mem": stats.get("Mem")
         }
         for record in content
-        if record.get("Metrics", {}).get("hackbench_avg_time")
+        for stats in record.get("ContainerStats")
+
+        # Filter statistics from Collector only, bounded to the timeframe when
+        # load was actually running. LoadStopTs is adjusted by 1 sec, since
+        # this timestamp is taken after the workload container was stopped, so
+        # that the actual load might have stopped slightly earlier.
+        if (stats.get("Name") == "collector" and
+            stats.get("Timestamp") > record.get("LoadStartTs") and
+            stats.get("Timestamp") < record.get("LoadStopTs") - timedelta(seconds=1))
```

Comment:
This list comprehension is pretty big now. It's fairly readable currently but I think it might be clearer if it was a set of conventional loops